plugins/inputs/t128_graphql/t128_graphql.go

```diff
@@ -0,0 +1,284 @@
+package t128_graphql
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Jeffail/gabs"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const (
+	//make this configurable?
+	DefaultRequestTimeout = time.Second * 5
+)
+
+type T128GraphQL struct {
+	CollectorName string            `toml:"collector_name"`
+	BaseURL       string            `toml:"base_url"`
+	EntryPoint    string            `toml:"entry_point"`
+	Fields        map[string]string `toml:"extract_fields"`
+	Tags          map[string]string `toml:"extract_tags"`
+
+	Client *http.Client
+	Query  string
+}
+
+var sampleConfig = `
+## Collect data using graphQL
+[[inputs.t128_graphql]]
+## Required. The telegraf collector name
+# collector_name = "arp-state"
+
+## Required. GraphQL ports vary across 128T versions
+# base_url = "http://localhost:31517"
+
+## The starting point in the graphQL tree for all configured tags and fields
+# entry_point = "allRouters[name:RTR_WEST_COMBO]/nodes/nodes[name:combo-west]/nodes/arp/nodes"
+
+## The fields to collect with the desired name as the key (left) and graphQL 
+## key as the value (right)
+# [inputs.t128_graphql.extract_fields]
+#   state = "state"
+
+## The tags for filtering data with the desired name as the key (left) and 
+## graphQL key as the value (right)
+# [inputs.t128_graphql.extract_tags]
+#   networkInterface = "networkInterface"
+#   deviceInterface = "deviceInterface"
+#   vlan = "vlan"
+#   ipAddress = "ipAddress"
+#   destinationMac = "destinationMac"
+`
+
+func (*T128GraphQL) SampleConfig() string {
+	return sampleConfig
+}
+
+func (*T128GraphQL) Description() string {
+	return "Read data from a graphQL query"
+}
+
+func (plugin *T128GraphQL) Init() error {
+	if plugin.CollectorName == "" {
+		return fmt.Errorf("collector_name is a required configuration field")
+	}
+
+	if plugin.BaseURL == "" {
+		return fmt.Errorf("base_url is a required configuration field")
+	}
+
+	if plugin.BaseURL[len(plugin.BaseURL)-1:] != "/" {
+		plugin.BaseURL += "/"
+	}
+
```

Comment:
aren't the entryPoint, fields and tags required as well for this whole thing to work?

Reply:
Good point. I have error handling to cover but it would be more clear to catch it up front. Entry-point and field are definitely required, not sure about tags -- @gregschrock ??

Reply:
As far as I know, from influx line protocol. Measurement name is required. Tags are optional. At least one field is required. https://docs.influxdata.com/influxdb/v1.8/write_protocols/line_protocol_tutorial/ . I'm not sure why people would be using this without tags, but I guess why not give them the freedom

Reply:
- [ ] require entry point and fields

Reply:
> guess why not give them the freedom

If its incomplete/invalid there is no value in having such freedom except for wasting CPU/Mem resources. Most users (like me) are just forgetful and would like to be told when something is wrong 😄